src/firebase.js

```diff
@@ -21,15 +21,31 @@ export { firebase };
 export const db = firebase.firestore();
 export const auth = firebase.auth();
 
-import { voyage } from "./store";
-const queryId = window.location.search.split("i=")?.[1];
+import { voyage, excursion } from "./store";
+
+let params = (new URL(window.location)).searchParams;
+const excursionId = params.get("e");
+const voyageId = params.get("v");
 let voyageRef;
-if (queryId) {
-  voyageRef = db.doc(`/voyages/${queryId}`);
+if (excursionId) {
+  voyageRef = db.doc(`/voyages/${voyageId}`);
 } else {
   voyageRef = db.collection("voyages").doc();
 }
 voyage.set(voyageRef);
+
+
+
+let excursionRef
+
+
+if (voyageId && excursionId) {
+  excursionRef = db.doc(`/voyages/${voyageId}/excursions/${excursionId}`);
+} else {
+  excursionRef = voyageRef.collection('excursions').doc();
+}
+excursion.set(excursionRef);
```

Comment:
this can instead be updated implicitly by including the dollar sign, it uses the actual value of the store, and is what you should be doing MOST of the time.

```js
if (voyageId && excursionId) {
   $excursion = db.doc(`/voyages/${voyageId}/excursions/${excursionId}`);
 } else {
   $excursion = voyageRef.collection('excursions').doc();
 }
```

Reply:
Haha ok. I did that because it's what your voyage code right above it did HEHE! I'll fix

Reply:
Uh @SvenWritesCode does that mean we need to change the file to a .svelte file?

Reply:
oh yeah, that must have been why I did it. it only works in svelte files